File: BE-HospitalAppointmentSchedule/HospitalAppointmentShedule.Domain/Models/Specialty.cs
﻿using System;
using System.Collections.Generic;

namespace HospitalAppointmentShedule.Domain.Models;

public partial class Specialty
{
    public int SpecialtyId { get; set; }

    public string SpecialtyName { get; set; } = null!;

    public string? SpecialtyDescription { get; set; }

    public string? Image { get; set; }

    public virtual ICollection<Service> Services { get; set; } = new List<Service>();

    public virtual ICollection<Doctor> Doctors { get; set; } = new List<Doctor>();
}

File: FE-HospitalAppointmentSchedule/app/src/main/java/com/example/project_prm392/model/Specialty.java
package com.example.project_prm392.model;

import java.util.List;

public class Specialty {
    private int specialtyId;
    private String specialtyName;
    private String description;
    private String image;
    private List<Doctor> doctors;
    private List<Service> services;

    // Constructor
    public Specialty() {
    }

    public Specialty(int specialtyId, String specialtyName, String description, String image) {
        this.specialtyId = specialtyId;
        this.specialtyName = specialtyName;
        this.description = description;
        this.image = image;
    }

    // Getters and Setters
    public int getSpecialtyId() {
        return specialtyId;
    }

    public void setSpecialtyId(int specialtyId) {
        this.specialtyId = specialtyId;
    }

    public String getSpecialtyName() {
        return specialtyName;
    }

    public void setSpecialtyName(String specialtyName) {
        this.specialtyName = specialtyName;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public List<Doctor> getDoctors() {
        return doctors;
    }

    public void setDoctors(List<Doctor> doctors) {
        this.doctors = doctors;
    }

    public List<Service> getServices() {
        return services;
    }

    public void setServices(List<Service> services) {
        this.services = services;
    }
} 